plugins/sharding/logical_enumerator.rb

```diff
@@ -63,9 +72,35 @@ def each_internal(order)
         end
       end
 
+      private
+      def each_with_index_internal(order)
+        index = 0
+        each_internal(order) do |shard, shard_range| 
+          yield(shard, shard_range, index)
+          index += 1
+        end
+      end
+
+      private
+      def compute_size
+        context = Context.instance
+        prefix = "#{@logical_table}_"
+        size = 0
+        context.database.each_name(:prefix => prefix) do |name|
+          shard_range_raw = name[prefix.size..-1]
+  
+          if shard_range_raw =~ /\A(\d{4})(\d{2})\z/ ||
+             shard_range_raw =~ /\A(\d{4})(\d{2})(\d{2})\z/
+            size += 1
+          end
+        end
+        size
+      end
+
       def each_shard_with_around(order)
         context = Context.instance
         prefix = "#{@logical_table}_"
+        unref_immediately = @options[:unref_immediately] || false
```

Comment:
`|| false` is redundant.
If `:unref_immediately` isn't specified, `@options[:unref_immediately]` returns `nil`. `nil` is one of a false value.

Reply:
Fixed.